astropy/units/tests/test_quantity_non_ufuncs.py

```diff
@@ -2033,6 +2034,53 @@ def test_eigvalsh(self):
         assert_array_equal(w, wx)
 
 
+class TestRecFunctions(metaclass=CoverageMeta):
+
+    def test_structured_to_unstructured(self):
+        # can't unstructure something with incompatible units
+        with pytest.raises(u.UnitConversionError, match="'m'"):
+            rfn.structured_to_unstructured(u.Quantity((0, 0.6), u.Unit("(eV, m)")))
+
+        # it works if all the units are equal
+        struct = u.Quantity((0, 0, 0.6), u.Unit("(eV, eV, eV)"))
+        unstruct = rfn.structured_to_unstructured(struct)
+        assert_array_equal(unstruct, [0, 0, 0.6] * u.eV)
+
+        # also if the units are convertible
+        struct = u.Quantity((0, 0, 0.6), u.Unit("(eV, eV, keV)"))
+        unstruct = rfn.structured_to_unstructured(struct)
+        assert_array_equal(unstruct, [0, 0, 600] * u.eV)
+
+        struct = u.Quantity((0, 0, 1.7827e-33), u.Unit("(eV, eV, g)"))
```

Comment:
Nice check on the equivalency!

Reply:
It was nice to realize it worked. I hadn't intentionally ensured it would.